DFC.Digital/DFC.Digital.Web.Sitefinity.DfcSearchModule/Service/DfcSearchService.cs

```diff
@@ -36,55 +40,121 @@ public DfcSearchService(ISearchService<JobProfileIndex> searchService, ISearchIn
 
         public override bool IndexExists(string indexName)
         {
-            if (!string.IsNullOrEmpty(indexName) && index.StartsWith(indexName, StringComparison.OrdinalIgnoreCase))
+            var activeIndex = string.Empty;
+            try
             {
-                return searchService.IndexExists(index);
+                if (!string.IsNullOrEmpty(indexName) && index.StartsWith(indexName, StringComparison.OrdinalIgnoreCase))
+                {
+                    activeIndex = index;
+                    return searchService.IndexExists(index);
+                }
+                else
+                {
+                    activeIndex = indexName;
+                    return base.IndexExists(indexName);
+                }
             }
-            else
+            catch (Exception exception)
             {
-                return base.IndexExists(indexName);
+                applicationLogger.Error($" Method - {MethodBase.GetCurrentMethod().Name} on index {activeIndex} failed with an exception", exception);
+                throw;
```

Comment:
You don't need to throw again mate, as the `applicationLogger.Error` will wrap the exception into logged exception and throws it back.